video_info_test.go

```diff
@@ -211,3 +217,10 @@ func TestThumbnail(t *testing.T) {
 func newDate(y, m, d int) time.Time {
 	return time.Date(y, time.Month(m), d, 0, 0, 0, 0, time.UTC)
 }
+
+func newClient(t *testing.T) Client {
+	return Client{
+		HTTPClient: http.DefaultClient,
+		Logger:     log.Logger,
```

Comment:
@zikaeroh I could not find an adapter to use `testing.T` as output for zerolog. I probably have to write a `io.Writer` that calls the `Log` function of a test case?

Reply:
Probably. I use zap (after hitting some bad zerolog behavior) and wrote this for it: https://github.com/hortbot/hortbot/blob/master/internal/pkg/testutil/testutil.go

Takes some tweaking to get line info correct since you have to make sure it's ignoring the writer's stack frame.

I'm away from anything but my phone, so I can't really look at this PR to give any feedback.

Reply:
Okay, done.

Reply:
Unfortunately the caller is getting lost:
```
=== RUN   TestThumbnail
--- PASS: TestThumbnail (0.85s)
    client_test.go:24: {"level":"debug","message":"Fetching https://www.youtube.com/watch?v=FrG4TEcSuRg"}
    client_test.go:24: {"level":"debug","message":"Fetching http://img.youtube.com/vi/FrG4TEcSuRg/default.jpg"}
    client_test.go:24: {"level":"debug","message":"Fetching http://img.youtube.com/vi/FrG4TEcSuRg/hqdefault.jpg"}
    client_test.go:24: {"level":"debug","message":"Fetching http://img.youtube.com/vi/FrG4TEcSuRg/maxresdefault.jpg"}
    client_test.go:24: {"level":"debug","message":"Fetching http://img.youtube.com/vi/FrG4TEcSuRg/mqdefault.jpg"}
    client_test.go:24: {"level":"debug","message":"Fetching http://img.youtube.com/vi/FrG4TEcSuRg/sddefault.jpg"}
```

Reply:
Yes, you'd need to use `CallerWithSkipFrameCount` on the logger you put in `testWriter` to ensure that the `Write` call's frame is skipped.